src/server/frontend_wayland/window_wl_surface_role.cpp

```diff
@@ -335,7 +344,9 @@ void mf::WindowWlSurfaceRole::commit(WlSurfaceState const& state)
             populate_spec_with_surface_data(spec());
         }
 
-        if (pending_changes)
+        apply_pending_size_bounds();
```

Comment:
This function name is misleading. It does two significant things:

1. consuming idempotent size bound changes
2. remembering the new size bounds so as to implement 1

The name does not reflect _either_ of these but the first is most important. The casual reader will not spot the side effect of changing `pending_changes->is_empty()`.

Maybe it should be called `consume_idempotent_size_bound_changes()` as 2 is an implementation detail?